doc/design/persistent_volumes_etcd_data.md

```diff
@@ -0,0 +1,125 @@
+# Use Persistent Volumes for etcd data
+
+## Proposal and Motivation
+
+Today etcd-operator creates ephemeral etcd members: in their pod the etcd data is stored inside a volume of type emptyDir and the pod restart policy is set to `Never`.
+
+This covers some failure cases but won't cover other failure cases.
+
+### Example cases
+
+Assume a N members etcd cluster with all members healthy
+
+#### Case A
+
+##### Event
+
+One of these events happens:
+* one etcd member process exits (processes crashes, killed etc...)
+* a k8s node executing one etcd member pod reboots and comes back before node controller pod eviction triggers
+
+##### Effect
+
+The pod will go in failed state, the etcd cluster remains quorate. Etcd-operator will remove the member from the etcd cluster and add a new member to the cluster and creates a new pod with a new name.
+
+#### Case B
+
+##### Event
+
+* a k8s node executing one etcd member pod becomes partitioned (network partitioning, goes down, etc...) and cannot talk to the api server, after some time the node controller will evict its pods.
+
+##### Effect
+
+The pod will go in an unknown/failed state, the etcd cluster remains quorate. Etcd-operator will remove the member from the etcd cluster and add a new member to the cluster and creates a new pod with a new name.
+
+#### Case C
+
+##### Event
+
+One of these events happens:
+* a majority of etcd processes exits (processes crashes, killed etc...) 
+* k8s nodes scheduling a majority of etcd members pods reboots and come back before node controller pod eviction triggers
+
+##### Effect
+
+The pods will go in failed state, the etcd cluster becomes unquorate. Etcd-operator can only restore the cluster from a backup.
+
+#### Case D
+
+##### Event
+
+* k8s nodes executing a majority of etcd members pods becomes partitioned (network partitioning, goes down, etc...) and cannot talk to the api server, after some time the node controller will evict their pods.
+
+##### Effect
+
+The pods will go in an unknown/failed state and eventually be removed from the API server, the etcd cluster becomes unquorate. Etcd-operator can only restore the cluster from a backup.
+
+
+#### Reasons
+
+Currently etcd operator handles cases A and B but when a cluster becomes unquorate (cases C and D) the current option is to restore the cluster from a backup.
+
+User may desire consistency over availability and prefer to wait for the cluster the return in a quorate state instead of restoring from a backup (that will contain old data).
+
+This proposal aims to fix cases C and D permitting to the cluster the return in a quorate state without the need to restore from a backup.
+
+## Proposed Changes
+
+#### Changes
+
+The change to be effective and cover the above failure scenario is split in two parts.
+
+### Part 1
```

Comment:
Some thoughts:

- Remove the above "Reasons" section. Add individual "Analysis" section to each part. For Part 1:
  ```
  For failure cases A and C, we can tolerate pod restart if we use PV to store etcd data.
  ```
- Remove "Problems" section. Cover problems & limitations in "Analysis".